deep_qa/data/token_indexers/token_indexer.py

```diff
@@ -75,7 +74,6 @@ def pad_token_sequence(self,
         """
         raise NotImplementedError
 
-    @staticmethod
-    def from_params(params: Params):
-        choice = params.pop_choice('type', list(token_indexers.keys()), default_to_first_choice=True)
-        return token_indexers[choice].from_params(params)
```

Comment:
One idea to fix this would be to have parameter checking only at the Model level. This would remove the need for the `from_params` on everything and we could just log parameters inside classes. This would have the benefit that the log would show where the parameter was defined, which it currently doesn't because all the logging happens in params. The replicability is nice, but we now achieve that with Dockerised experiments, so maybe it is less useful.